src/components/select/SelectWithFilter.tsx

```diff
@@ -91,13 +102,18 @@ export const selectWithFilter = (Component: (React.ComponentClass<ISelectWithFil
                     <FilterBoxConnected
                         id={this.props.id}
                         containerClasses={[filterBoxContainerClassNames]}
-                        onKeyDown={(this.props as any).onKeyDown}
-                        onKeyUp={(this.props as any).onKeyUp}
+                        onKeyDown={(e: React.KeyboardEvent<HTMLElement>) => callIfDefined((this.props as any).onKeyDown, e)}

```

Comment:
Why the as any here?

Reply:
this kind of structure doesnt know if the Parent component has or not the prop onKeyUp.
We need to set the props as any to call onKeyUp. Depending of what kind of parent you use, you can have it or not

Reply:
can't you just add it as optional in `ISelectWithPredicateAndFilterProps` instead? same for the onKeyUp prop?